### PR TITLE
tests: net: increase stack size for code coverage

### PR DESCRIPTION
**Pull request by keylime-unicorn**
This is a partial fix for #17323

Fixes issues with tests/net/lib/dns_addremove

Signed-off-by: Cami Carballo <cami.carballo@intel.com>

### CODE REVIEW

File: tests/net/lib/dns_addremove/prj.conf [hunk 1]
```diff
@@ -9,12 +9,13 @@ CONFIG_DNS_NUM_CONCUR_QUERIES=1
...
 CONFIG_NET_MGMT_EVENT_THREAD_PRIO=7
 CONFIG_NET_MGMT_EVENT_QUEUE_SIZE=2
 CONFIG_NET_IPV4=y
 CONFIG_NET_IPV6=y
 CONFIG_NET_ARP=n
 CONFIG_PRINTK=y
 CONFIG_ZTEST=y
-CONFIG_MAIN_STACK_SIZE=1344
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_ZTEST_STACKSIZE=2048
```
Comment (andrewboie) [inline tests/net/lib/dns_addremove/prj.conf]: Better change: update subsys/testsuite/Kconfig default if coverage is enabled

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/net/lib/dns_addremove/prj.conf [hunk 2]
```diff
@@ -9,12 +9,10 @@ CONFIG_DNS_NUM_CONCUR_QUERIES=1
...
 CONFIG_NET_MGMT_EVENT=y
-CONFIG_NET_MGMT_EVENT_STACK_SIZE=768
 CONFIG_NET_MGMT_EVENT_THREAD_PRIO=7
 CONFIG_NET_MGMT_EVENT_QUEUE_SIZE=2
 CONFIG_NET_IPV4=y
 CONFIG_NET_IPV6=y
 CONFIG_NET_ARP=n
 CONFIG_PRINTK=y
 CONFIG_ZTEST=y
-CONFIG_MAIN_STACK_SIZE=1344
```
Comment (andrewboie) [inline tests/net/lib/dns_addremove/prj.conf]: Hmmm dns_addremove test is now failing in CI. But the main stack size shouldn't matter, it runs under ztest anyway. We also ought to be getting stack overflow exceptions. Need to look into this further.

Comment (camicarballo) [reply tests/net/lib/dns_addremove/prj.conf]: The dns_addremove test is failing with a 'recursive spinlock' error apparently.

**Review by jukkar — APPROVED** [review]

**Comment by pfalcon** [comment]
@camicarballo, Please rebase on the latest master.

**Review by pfalcon — APPROVED** [review]

**Comment by andrewboie** [comment]
dns_addremove tests seem to be failing.

**Pull request merged**